src/coinbase/wallet.ts

```diff
@@ -361,8 +341,17 @@ export class Wallet {
    *
    * @returns The list of Addresses.
    */
-  public listAddresses(): Address[] {
-    return this.addresses;
+  public async listAddresses(): Promise<WalletAddress[]> {
```

Comment:
If `this.addresses` is already set, this can also short-circuit and return the cached/memoized value?

Reply:
Sure, I thought it shouldn't cache any data since it's a public method. However, cache validation seems like a more difficult problem to tackle :)